Add docs for smart retry (#1939)
saucelabs/sauce-docs: docs/web-apps/automated-testing/cypress/yaml/v1.md

  - name: My Saucy Test
    passThreshold: 2
```

---

### `smartRetry`

<p><small>| OPTIONAL | OBJECT |</small></p>

> Specifies the retry strategy to apply for that suite. It should be used along with [retries](#retries).

Reviewer: ```suggestion
Specifies the retry strategy to apply for that suite. Requires [retries](#retries) to be >= 1.
```

```yaml
sauce:
  retries: 3
suite:
  - name: My Saucy Test
    smartRetry:
      failedOnly: true
```

---

#### `failedOnly`

<p><small>| OPTIONAL | BOOLEAN |</small></p>

> When set to `true`, the retry will test only the tests that failed during the previous try.

Reviewer: ```suggestion
When set to `true`, only the tests that failed during the previous attempt are retried.
```
